### PR TITLE
applications: nrf5340_audio: Refactor pscm

### PR DESCRIPTION
**Pull request by koffes**
Moved the PCM stream channel modifier library

Signed-off-by: Kristoffer Rist Skøien <kristoffer.skoien@nordicsemi.no>

### CODE REVIEW

**Comment by koffes** [comment]
I'd be glad to get feedback on especially the new `audio_defines.h` It has been included by `<>` in other files as it is a global header.

File: lib/pcm_stream_channel_modifier/Kconfig [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menuconfig PSCM
+	bool "PSCM - PCM stream channel modifier library"
```
Comment (erikrobstad) [inline lib/pcm_stream_channel_modifier/Kconfig:8]:
```suggestion
	bool "PSCM - PCM Stream Channel Modifier library"
```

The .rst file uses capitalized letters

File: lib/pcm_stream_channel_modifier/Kconfig [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+	help
+	  Library for manipulating Pulse coded modulation (PCM) streams
```
Comment (erikrobstad) [inline lib/pcm_stream_channel_modifier/Kconfig:10]:
```suggestion
	  Library for manipulating pulse coded modulation (PCM) streams
```

File: lib/pcm_stream_channel_modifier/Kconfig (hunk 1 above)
Comment (erikrobstad) [reply lib/pcm_stream_channel_modifier/Kconfig:8]: Same on line 15

File: tests/lib/pcm_stream_channel_modifier/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#
+# Copyright (c) 2018 Nordic Semiconductor ASA
```
Comment (erikrobstad) [inline tests/lib/pcm_stream_channel_modifier/CMakeLists.txt:2]: 2018 = 2023?

**Review by erikrobstad — APPROVED** [review]

File: tests/lib/pcm_stream_channel_modifier/CMakeLists.txt (hunk 3 above)
Comment (koffes) [reply tests/lib/pcm_stream_channel_modifier/CMakeLists.txt:2]: This file originates from 2018, so I kept it.

File: doc/nrf/libraries/others/pcm_stream_channel_modifier.rst [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+PCM Stream Channel Modifier
+###########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+PCM Stream Channel Modifier library enables users to split pulse-code modulation (PCM) streams from stereo to mono
+or combine mono streams to form a stereo stream.
```
Comment (greg-fer) [inline doc/nrf/libraries/others/pcm_stream_channel_modifier.rst:11]:
```suggestion
PCM Stream Channel Modifier library enables users to split pulse-code modulation (PCM) streams from stereo to mono or combine mono streams to form a stereo stream.
```
We have a one-sentence-per-one-line rule in the NCS docs.

File: include/pcm_stream_channel_modifier.h [hunk 5]
```diff
@@ -8,8 +8,7 @@
 #define _PCM_STREAM_CHANNEL_MODIFIER_H_
 
 #include <zephyr/kernel.h>
```
Comment (greg-fer) [inline include/pcm_stream_channel_modifier.h:10]: The doxygen defines are missing here. Check the headers of the libraries that you moved previously for reference.

Comment (greg-fer) [reply include/pcm_stream_channel_modifier.h:10]: (Also, adding them means adding the closing define }@ tags at the end of the header.)

**Review by greg-fer — CHANGES_REQUESTED** [review]
See comments for requested changes.

**Pull request merged**